crates/holochain_p2p/src/types/mock_network.rs

```diff
@@ -1,516 +0,0 @@
-                let msg = crate::wire::WireMessage::decode(data.as_ref()).unwrap();
-                HolochainP2pMockMsg::Wire { to_agent, msg, dna }
-            }
-            kwire::Wire::Broadcast(kwire::Broadcast {
-                to_agent,
-                data,
-                space,
-                ..
-            })
-            | kwire::Wire::DelegateBroadcast(kwire::DelegateBroadcast {
-                to_agent,
-                data,
-                space,
-                ..
-            }) => {
-                let to_agent = holo_hash::AgentPubKey::from_kitsune(&to_agent);
-                let dna = holo_hash::DnaHash::from_kitsune(&space);
-                match data {
-                    BroadcastData::User(data) => {
-                        let msg = crate::wire::WireMessage::decode(data.as_ref()).unwrap();
-                        HolochainP2pMockMsg::Wire { to_agent, msg, dna }
-                    }
-                    BroadcastData::AgentInfo(info) => HolochainP2pMockMsg::PublishedAgentInfo {
-                        to_agent,
-                        dna,
-                        info,
-                    },
-                    BroadcastData::Publish { .. } => {
-                        todo!()
-                    }
-                }
-            }
-            kwire::Wire::Gossip(kwire::Gossip {
-                data,
-                space,
-                module,
-            }) => {
-                use kitsune_p2p::gossip::sharded_gossip::*;
-                use kitsune_p2p_types::codec::Codec;
-                let gossip = match module {
-                    GossipModuleType::ShardedRecent | GossipModuleType::ShardedHistorical => {
-                        GossipProtocol::Sharded(
-                            ShardedGossipWire::decode_ref(data.as_ref()).unwrap().1,
-                        )
-                    }
-                };
-                let dna = holo_hash::DnaHash::from_kitsune(&space);
-                HolochainP2pMockMsg::Gossip {
-                    module,
-                    dna,
-                    gossip,
-                }
-            }
-            kwire::Wire::MetricExchange(msg) => HolochainP2pMockMsg::MetricExchange(msg),
-            kwire::Wire::PeerGet(msg) => HolochainP2pMockMsg::PeerGet(msg),
-            kwire::Wire::PeerGetResp(msg) => HolochainP2pMockMsg::PeerGetResp(msg),
-            kwire::Wire::PeerQuery(msg) => HolochainP2pMockMsg::PeerQuery(msg),
-            kwire::Wire::PeerQueryResp(msg) => HolochainP2pMockMsg::PeerQueryResp(msg),
-            kwire::Wire::PeerUnsolicited(msg) => HolochainP2pMockMsg::PeerUnsolicited(msg),
-            kwire::Wire::CallResp(msg) => HolochainP2pMockMsg::CallResp(msg.data),
-            kwire::Wire::Failure(msg) => HolochainP2pMockMsg::Failure(msg.reason),
-            kwire::Wire::FetchOp(_) => todo!(),
-            kwire::Wire::PushOpData(_) => todo!(),
-        }
-    }
-}
```

Comment:
@neonphog this is the mock network that Tom wrote, I always intended to some day learn from it and maybe revive or reuse it, maybe you want to do that

Reply:
The concept is straight forward... but I imagine the practice of writing tests that include requiring the knowledge of constructing all the response messages and in the right order would be long-winded and difficult to maintain.

I guess this will remain in our commit history if we ever want to look at it.